### DISCUSSION TITLE
Publish a package using bun

### DISCUSSION THREAD

**Discussion by infodusha** [oven-sh/bun#5365] [Show and tell]
So basically when I did migration of my packages I has issues when publishing.
Since bun has no publish command you actually have to install npm: `bun add -g npm`
Therefore, you need to auth and that may be not that clear in CI - `NODE_AUTH_TOKEN` env no more works when running npm with bun.
I found a solution in the bun codebase.
What you need it to add `.npmrc` with following content:
```
//registry.npmjs.org/:_authToken=${NODE_AUTH_TOKEN}
```
So you can read auth token from env again and everything works.

**Comment by tomchen** [comment]
Now bun supports publish but you will likely still fail in CI publishing - that is because bun uses `NPM_CONFIG_TOKEN` and does not support `NODE_AUTH_TOKEN`. So the solution is to simply replace `NODE_AUTH_TOKEN` by `NPM_CONFIG_TOKEN`

(and you don't need `echo "//registry.npmjs.org/:_authToken=\${NODE_AUTH_TOKEN}" > ~/.npmrc` as suggested here or by ChatGPT/GitHub Copilot, this solution is outdated)

https://github.com/oven-sh/bun/issues/14824
https://bun.sh/docs/cli/publish
